app/src/main/java/com/getsensibill/spendmanager/demo/fragment/FragmentJavaActivity.java

```diff
@@ -87,4 +83,68 @@ public void onDisplayNetworkError(boolean networkNotAvailable) {
     public void onRequestFinish(@NotNull UiFinishReason uiFinishReason) {
         finish();
     }
+
+    /**
+     * Required Override when using [WebUiNetworkErrorFragment.Listener]
+     * <p>
+     * This is the returning call from the [WebUiNetworkErrorFragment] when the ok/close button is
+     * pressed. If using a fragment integration and not subclassing [WebUiActivity], then the
+     * error view will not know how to handle the event.
+     */
+    @Override
+    public void onNetworkErrorClose() {
+        onRequestFinish(UiFinishReason.UNABLE_TO_LOAD_SPA);
+    }
+
+    // -----WebUiNetworkErrorFragment.Listener Overrides -----
+
+    /**
+     * Required Override when using [WebUiNetworkErrorFragment.Listener]
+     * <p>
+     * This is the returning call from the [WebUiNetworkErrorFragment] when the retry button is
+     * pressed. If using a fragment integration and not subclassing [WebUiActivity], then the
+     * error view will not know how to handle the event.
+     */
+    @Override
+    public void onNetworkErrorRetry() {
+        loadWebUi();
+    }
+
+    private void loadWebUi() {
+        if (this.isFinishing()) {
+            return;
+        }
+        // reuses the fragment if it already existed
+        if (getFragment(TAG_WEB_FRAGMENT) != null) {
+            setFragment((WebUiFragment) getFragment(TAG_WEB_FRAGMENT), TAG_WEB_FRAGMENT);
+            return;
+        }
+
+        WebUiFragment fragment = new WebUiFragment();
+        Bundle bundle = new Bundle();
+
+        // Pass in a navigation override. Defaults as .DASHBOARD
+        bundle.putParcelable(
+                WebUiFragment.ARG_NAVIGATION_OVERRIDE,
+                NavigationIntent.DASHBOARD.INSTANCE
+        );
+
+        // Pass in a custom Theme override. Defaults to null
+        WebTheme theme = new WebTheme(new Brand());
+        bundle.putParcelable(WebUiFragment.ARG_WEB_THEME_OVERRIDE, theme);
+
+        fragment.setArguments(bundle);
+        setFragment(fragment, TAG_WEB_FRAGMENT);
+    }
+
+    private void setFragment(SbBaseFragment fragment, String TAG) {
```

Comment:
These methods could probably use better names, maybe `showFragmentInContainer()` and for the other one, we could just use the name of the `FragmentManager` method: `findFragmentByTag()`

Reply:
also, the `tag` fragment argument shouldn't be capitalized

Reply:
updated